lib/worker.js

```diff
@@ -330,12 +345,47 @@ class AssetComputeWorker {
         }
     }
 
+    sendEventsBeforeActionTimeout() {
+        return setTimeout(
+            async () => {
+                console.log(`Action is about to timeout in ${timeUntilTimeout()}ms. Sending Adobe I/O events before timeout.`);
+                const errorMessage = `Worker timed out without result after ${this.timers.actionDuration.end()} seconds`;
+                // ensure failure events are sent for any non successful rendition before timeout
+                if(this.renditions){
+                    for (const rendition of this.renditions) {
+                        if (!rendition.eventSent) {
+                            await this.events.sendEvent(EVENT_RENDITION_FAILED, {
+                                rendition: rendition.instructionsForEvent(),
+                                errorReason: Reason.GenericError,
+                                errorMessage: errorMessage,
+                                activationIds: this.params.customWorker ? [process.env.__OW_ACTIVATION_ID] : undefined
+                            });
+                            rendition.eventSent = true;
+                        }
+                    }
+                }
+
+                await this.metrics.handleError(new GenericError(errorMessage), {
```

Comment:
Are we still sending the `timeout` metric? I think we must ensure this somehow as we would fully consider this path as a timeout. Maybe send it manually here instead of the error metric.

Currently it happens here: https://github.com/adobe/node-openwhisk-newrelic/blob/af62c5ce3e3c465655e624b73d81d290ed39e03e/lib/newrelic.js#L180

Reply:
So the answer is currently not typically, because the timeout metric sends with 5s left and we start timing out with 15s left. I thought this error metric was supposed to replace the timeout metric. Instead of marking it as an `error` metric, we can define it as a `timeout` metric.

Reply:
renamed the `error` metric sent here to a `timeout` metric.